src/Simulation/DepletionVoltage.jl

```diff
@@ -0,0 +1,164 @@
+        depleted = ϕmax - ϕmin < abs(U)
+        if (initial_depletion && !depleted) || (!initial_depletion && depleted)
+            if T(U)>=0
+                start_local_search = T(U)
+            else
+                start_local_search = T(U)
+            end
+            break
+        end
+    end
+    local_range::AbstractRange = range(start_local_search, last(potential_range), step = step(potential_range))
+    size_data = size(sim.point_types.data)
+    
+    if size_data[2] == 1
+        Ix = CartesianIndex(1,0,0)
+        Iz = CartesianIndex(0,0,1)
+        neighbours = [Ix,-Ix,Iz,-Iz]
+    else
+        Ix = CartesianIndex(1,0,0)
+        Iy = CartesianIndex(0,1,0)
+        Iz = CartesianIndex(0,0,1)
+        neighbours = [Ix,-Ix,Iy,-Iy,Iz,-Iz]
+    end
+
+    scale = start_local_search
+    undepleted = 0
+    for (i,idx) in enumerate(inside)
+        if !(0 in [size_data[j] == 1 || idx[j] != 1 && idx[j] != size_data[j] for j in 1:length(idx)])#idx[1]!=1 && idx[1]!=size_data[1] && idx[2]!=1 && idx[2]!=size_data[2] && idx[3]!=1 && idx[3]!=size_data[3]
+            for scale_loc in local_range
+                center_pot = T(scale_loc) * sim.weighting_potentials[contact_id].data[idx] + ϕρ[idx]
+                min_pot = T(scale_loc) * sim.weighting_potentials[contact_id].data[idx-Ix] + ϕρ[idx-Ix]
+                max_pot = min_pot
+                for neighbour in neighbours
+                    n_idx = idx + neighbour
+                    local_pot = T(scale_loc) * sim.weighting_potentials[contact_id].data[n_idx] + ϕρ[n_idx]
+                    if local_pot < min_pot
+                        min_pot = local_pot
+                    elseif local_pot > max_pot
+                        max_pot = local_pot
+                    end
+                end
+                if !(min_pot>center_pot || max_pot<center_pot)
+                    if abs(scale_loc)> abs(scale)
+                        scale = T(scale_loc)
+                        local_range = range(scale, last(potential_range), step = step(potential_range))
+                    end         
+                    break
+                end
+                if scale_loc == local_range[end]
+                    undepleted+=1
+                end         
+            end
+        else
+            deleteat!(inside,i) #Delete indices which are in bulk bit, but still have non defined neighbour (Maybe we can get rid of this, I just kept it for safety)
+        end
+    end
+    if initial_depletion
+        depletion_voltage = potential_range[1]
+    elseif undepleted == 0
+        depletion_voltage = scale
+    end
+    
+    if verbose
+        if !isnan(depletion_voltage)
+            @info "The depletion voltage of the detector is ($(depletion_voltage) ± $(T(step(potential_range)))) V applied to contact $(contact_id)."
+        else
+            @warn "The depletion voltage is not in the specified range $(extrema(potential_range).*u"V")."
+        end
```

Comment:
... this message is printed out (that it is **NOT** in the range).

Then, I would also remove the tests for which detectors with no impurity density yield a depletion voltage of 0V.